pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/TableUpsertMetadataManagerFactory.java

```diff
@@ -35,8 +32,7 @@ private TableUpsertMetadataManagerFactory() {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(TableUpsertMetadataManagerFactory.class);
 
-  public static TableUpsertMetadataManager create(TableConfig tableConfig, Schema schema,
-      TableDataManager tableDataManager, ServerMetrics serverMetrics) {
+  public static TableUpsertMetadataManager create(TableConfig tableConfig) {
```

Comment:
Suggest not making this change. For plugin factories, `init()` is usually called within the factory

Reply:
kk. I made this change because isUpsertEnabled() is called in addSegment() to handle upsert specific states. So need to set `_tableUpsertMetadataManager` before preloading segments which happens as part of init() method. lemme use a boolean var for here.
```
public boolean isUpsertEnabled() {
    return _tableUpsertMetadataManager != null;
  }

public void addSegment(ImmutableSegment immutableSegment) {
    if (isUpsertEnabled()) {
      handleUpsert(immutableSegment);
      return;
    }
...
```